agent/api/task.go

```diff
@@ -110,6 +111,13 @@ type Task struct {
 	// used to look up the credentials for task in the credentials manager
 	credentialsID     string
 	credentialsIDLock sync.RWMutex
+
+	// cgroupSpecLock to reliably update the cgroup spec
+	cgroupSpecLock sync.RWMutex
+
+	// CgroupSpec attribute to capture the cgroup spec
+	// TODO: Inspect upon model changes
+	CgroupSpec *cgroup.Spec `json:omitempty`
```

Comment:
@samuelkarp I'm curious -- why isn't this sufficient? The task def isn't going to be platform-dependent.